src/Composer/Installers/TaoInstaller.php

```diff
@@ -6,7 +6,25 @@
  */
 class TaoInstaller extends BaseInstaller
 {
+    const EXTRA_TAO_EXTENSION_NAME = 'tao-extension-name';
+
     protected $locations = array(
         'extension' => '{$name}'
     );
+    
+    public function inflectPackageVars($vars)
+    {
+        $extra = $this->package->getExtra();
+
+        if (array_key_exists(self::EXTRA_TAO_EXTENSION_NAME, $extra)) {
+            $vars['name'] = $extra[self::EXTRA_TAO_EXTENSION_NAME];
+            return $vars;
+        }
+
+        $vars['name'] = str_replace('extension-', '', $vars['name']);
+        $vars['name'] = str_replace('-', ' ', $vars['name']);
+        $vars['name'] = lcfirst(str_replace(' ', '', ucwords($vars['name'])));
```

Comment:
This changes violate backward compatibility. May be replace this on `parent::inflectPackageVars`?

Reply:
But these manipulations are actual only for TAO extensions. I guess no need to change the default behavior in the BaseInstaller. And this installer has not been used anywhere else because we use own custom plugin, so BC definitely won't be broken.